From 5ffcd42a9af340764f9bd0ce3f50b156840c196a Mon Sep 17 00:00:00 2001
From: Ryszard Knop <ryszard.knop@intel.com>
Date: Fri, 31 Jul 2026 16:45:08 +0200
Subject: [PATCH 10105/15333] scripts/kernel-doc: Fix kdoc for Python 3.9-3.11

The syntax used by the excess description suggestion change works on
Python 3.12+, while we need to support 3.9+.

Signed-off-by: Ryszard Knop <ryszard.knop@intel.com>
Tested-by: Akira Yokosawa <akiyks@gmail.com>
Fixes: d7758384ccb4 ("scripts/kernel-doc: Suggest possible names for excess descriptions")
Reported-by: Akira Yokosawa <akiyks@gmail.com>
Closes: https://lore.kernel.org/22a9276d-c103-4306-a617-7f34abcd5c29@gmail.com/
Signed-off-by: Jonathan Corbet <corbet@lwn.net>
Message-ID: <20260731144508.912049-1-ryszard.knop@intel.com>
---
 tools/lib/python/kdoc/kdoc_parser.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tools/lib/python/kdoc/kdoc_parser.py b/tools/lib/python/kdoc/kdoc_parser.py
index 884f42584667..9cba20f827c4 100644
--- a/tools/lib/python/kdoc/kdoc_parser.py
+++ b/tools/lib/python/kdoc/kdoc_parser.py
@@ -601,7 +601,8 @@ class KernelDoc:
         if not suggestions:
             return ""
 
-        return f"(did you mean one of: '{"', '".join(suggestions)}')"
+        joined_suggestions = "', '".join(suggestions)
+        return f"(did you mean one of: '{joined_suggestions}')"
 
     def check_sections(self, ln, decl_name, decl_type):
         """
-- 
2.55.0

